vllm/_ipex_ops.py

```diff
@@ -396,10 +394,14 @@ def scaled_fp8_quant(
             assert num_token_padding is None, \
                 "padding not supported if output passed in"
             assert output.dtype == out_dtype
-        assert scale is None, "only dynamic fp8 quantization supported on XPU"
         assert not use_per_token_if_dynamic, (
             "per token dynamic fp8 quantization not supported on XPU")
-        scale = torch.zeros(1, device=input.device, dtype=torch.float32)
-        torch.ops.torch_ipex.dynamic_scaled_fp8_quant(output, input, scale)
+
+        if scale is None:
+            scale = torch.zeros(1, device=input.device, dtype=torch.float32)
+            torch.ops.torch_ipex.dynamic_scaled_fp8_quant(output, input, scale)
+        else:
+            assert scale.numel() == 1, f"{scale.shape}"
+            torch.ops.torch_ipex.static_scaled_fp8_quant(output, input, scale)
```

Comment:
so is this path tested? if not, I think we should not expose it.

Reply:
oh, this is used in q quant instead of kv cache quant. will revert .